File: SDK-64bit/doc/_downloads/almotion_walkto.cpp
#include <iostream>
#include <alproxies/almotionproxy.h>

int main(int argc, char **argv)
{
  if (argc < 2) {
    std::cerr << "Usage: motion_walkto pIp"
              << std::endl;
    return 1;
  }
  const std::string pIp = argv[1];

  AL::ALMotionProxy motion(pIp);

  motion.setStiffnesses("Body", 1.0f);
  motion.walkInit();

  // Example showing the walkTo command
  // as length of path is less than 0.4m
  // the path will be an SE3 interpolation
  // The units for this command are meters and radians
  float x  = 0.2f;
  float y  = 0.2f;
  // pi/2 anti-clockwise (90 degrees)
  float theta = 1.5709f;
  motion.walkTo(x, y, theta);
  // Will block until walk Task is finished

  // Example showing the walkTo command
  // as length of path is more than 0.4m
  // the path will be follow a dubins curve
  // The units for this command are meters and radians
  x  = -0.1f;
  y  = -0.7f;
  theta  = 0.0f;
  motion.walkTo(x, y, theta);

  return 0;
}

File: SDK-64bit/doc/_downloads/almotion_walkto.py
import sys
import math
from naoqi import ALProxy

if (len(sys.argv) < 2):
    print "Usage: 'python motion_walkto.py IP [PORT]'"
    sys.exit(1)

IP = sys.argv[1]
PORT = 9559
if (len(sys.argv) > 2):
    PORT = sys.argv[2]
try:
    proxy = ALProxy("ALMotion", IP, PORT)
except Exception,e:
    print "Could not create proxy to ALMotion"
    print "Error was: ",e
    sys.exit(1)

proxy.stiffnessInterpolation("Body", 1.0, 0.1)
proxy.walkInit()

# Example showing the walkTo command 
# The units for this command are meters and radians 
x  = 0.2 
y  = 0.2 
theta  = math.pi/2
proxy.walkTo(x, y, theta) 
# Will block until walk Task is finished

########
# NOTE #
########
# If walkTo() method does nothing on the robot,
# please refer to special walk protection in the
# setWalkTargetVelocity() method description below.